### PR TITLE
fix: fix error to status code mapping

### PR DESCRIPTION
**Pull request by rahmatrhd**

### CODE REVIEW

File: api/handler/v1beta1/appeal.go [hunk 1]
```diff
@@ -96,21 +97,45 @@ func (s *GRPCServer) CreateAppeal(ctx context.Context, req *guardianv1beta1.Crea
...
+			errors.Is(err, appeal.ErrInvalidResourceType),
+			errors.Is(err, appeal.ErrAppealInvalidExtensionDuration),
+			errors.Is(err, appeal.ErrGrantNotEligibleForExtension),
+			errors.Is(err, domain.ErrFailedToGetApprovers),
+			errors.Is(err, domain.ErrApproversNotFound),
+			errors.Is(err, domain.ErrUnexpectedApproverType),
+			errors.Is(err, domain.ErrInvalidApproverValue):
+			return nil, status.Errorf(codes.FailedPrecondition, err.Error())
+		default:
+			return nil, s.internalError(ctx, "failed to create appeal(s): %v", err)
```
Comment (bsushmith) [inline api/handler/v1beta1/appeal.go:130]: Can we also log these errors along with details?

File: api/handler/v1beta1/grant.go [hunk 2]
```diff
@@ -255,7 +255,7 @@ func (s *GRPCServer) ListUserRoles(ctx context.Context, req *guardianv1beta1.Lis
 
 	roles, err := s.grantService.ListUserRoles(ctx, user)
 	if err != nil {
-		return nil, status.Error(codes.Internal, "Internal Error")
+		return nil, s.internalError(ctx, "Internal Error")
```
Comment (bsushmith) [inline api/handler/v1beta1/grant.go:258]:
```suggestion
		return nil, s.internalError(ctx, "Internal Error : %s", err)
```

File: core/appeal/service.go [hunk 3]
```diff
@@ -1040,31 +1032,31 @@ func (s *Service) GrantAccessToProvider(ctx context.Context, a *domain.Appeal, o
...
-	extensionDurationRule, err := time.ParseDuration(AllowActiveAccessExtensionIn)
+	extensionDurationRule, err := time.ParseDuration(allowActiveAccessExtensionIn)
 	if err != nil {
-		return fmt.Errorf("%w: %v: %v", ErrAppealInvalidExtensionDuration, AllowActiveAccessExtensionIn, err)
+		return fmt.Errorf("%w: %q: %v", ErrAppealInvalidExtensionDuration, allowActiveAccessExtensionIn, err)
 	}
 
 	if !activeGrant.IsEligibleForExtension(extensionDurationRule) {
-		return ErrGrantNotEligibleForExtension
+		return fmt.Errorf("%w: extension is allowed %q before grant expiration", ErrGrantNotEligibleForExtension, allowActiveAccessExtensionIn)
```
Comment (bsushmith) [inline core/appeal/service.go:1059]: Would this value `allowActiveAccessExtensionIn` be in `hrs` or `days`?

File: core/approval/errors.go [hunk 4]
```diff
@@ -1,35 +0,0 @@
...
-	ErrAppealDuplicate                = errors.New("appeal with the same resource and role already exists")
-	ErrAppealInvalidExtensionDuration = errors.New("invalid appeal extension duration")
-	ErrAppealFoundActiveGrant         = errors.New("user still have an active grant")
-	ErrGrantNotEligibleForExtension   = errors.New("existing grant is not eligible for extension")
-	ErrCannotCreateAppealForOtherUser = errors.New("creating appeal for other individual user (account_type=\"user\") is not allowed")
-
-	ErrApprovalDependencyIsBlocked = errors.New("found previous approval step that is still in blocked")
-	ErrApprovalDependencyIsPending = errors.New("found previous approval step that is still in pending")
-	ErrApprovalStatusApproved      = errors.New("approval already approved")
-	ErrApprovalStatusRejected      = errors.New("approval already rejected")
```
Comment (bsushmith) [inline core/approval/errors.go:26]: All of these errors in this file are not required?

Comment (rahmatrhd) [reply core/approval/errors.go:26]: yes, looks like this is copied from `core/appeal/errors.go` but not sure why it was copied

File: core/appeal/service.go (hunk 3 above)
Comment (rahmatrhd) [reply core/appeal/service.go:1059]: this is in the golang duration format so it'll be in hours

File: api/handler/v1beta1/appeal.go (hunk 1 above)
Comment (rahmatrhd) [reply api/handler/v1beta1/appeal.go:130]: do you mean to also log them in the application? and what do you mean by "details"?

**Review by bsushmith — APPROVED** [review]

**Pull request merged**